project4/3700crawler

```diff
@@ -134,21 +174,56 @@ class WebCrawler:
             exit(1)
         self.session = match.group(1)
 
-        self.set_cookie()
-
-        login_resp = self.POST(LOGIN_FORM, make_formstr({
+        login_resp = self.POST(LOGIN_FORM, LOGIN_FORM, make_formstr({
             'username': self.username,
             'password': self.password,
             'csrfmiddlewaretoken': self.csrf,
             'submit': 'Log in',
-            'next': '/fakebook/'
+            'next': r'%2Ffakebook%2F'
         }))
         print(login_resp)
+        match = re.search('[Ss]et-[Cc]ookie:\W*sessionid=(\w+);', login_resp)
+        if (not match):
+            print('Couldn\'t extract SessionID token, aborting')
+            print(login_resp)
+            exit(1)
+        self.session = match.group(1)
+
+        return self.traverse_to(get_location(login_resp))
+
+def get_linkset(origin, page):
```

Comment:
`page` will be a dict of the headers and HTML now, so this function will need to change in order to merge with what's on `master`. I think it's just a small fix, though. You just need to search `page['HTML']` instead.